packages/components/pagination/src/use-pagination.ts

```diff
@@ -221,11 +225,7 @@ export function usePagination(originalProps: UsePaginationProps) {
     }
   }
 
-  function scrollTo(value: number, skipAnimation: boolean) {
-    const map = getItemsRefMap();
-
-    const node = map.get(value);
-
+  function scrollTo(node: HTMLElement | undefined, skipAnimation: boolean) {
```

Comment:
in `scrollTo` function, there is no logic for `node = undefined` case since it always returns in L229. Therefore, rewrite as the following

```diff
- function scrollTo(node: HTMLElement | undefined, skipAnimation: boolean) {
+ function scrollTo(node: HTMLElement, skipAnimation: boolean) {
-   if (!node || !cursorRef.current) return;
+   if (!cursorRef.current) return;
```

then trigger scrollTo when node is not undefined

```diff
- if (activePage && !disableAnimation && isActiveVisible) {
+ if (node && activePage && !disableAnimation && isActiveVisible) {
  scrollTo(node, activePage === activePageRef.current);
}
```